libcove/lib/common.py

```diff
@@ -29,23 +29,29 @@
 
 uniqueItemsValidator = validator.VALIDATORS.pop("uniqueItems")
 LANGUAGE_RE = re.compile("^(.*_(((([A-Za-z]{2,3}(-([A-Za-z]{3}(-[A-Za-z]{3}){0,2}))?)|[A-Za-z]{4}|[A-Za-z]{5,8})(-([A-Za-z]{4}))?(-([A-Za-z]{2}|[0-9]{3}))?(-([A-Za-z0-9]{5,8}|[0-9][A-Za-z0-9]{3}))*(-([0-9A-WY-Za-wy-z](-[A-Za-z0-9]{2,8})+))*(-(x(-[A-Za-z0-9]{1,8})+))?)|(x(-[A-Za-z0-9]{1,8})+)))$") # noqa
-validation_error_template_lookup = {'date-time': 'Date is not in the correct format',
-                           'uri': 'Invalid \'uri\' found',
-                           'string': '\'{}\' is not a string. Check that the value {} has quotes at the start and end. Escape any quotes in the value with \'\\\'', # noqa
-                           'integer': '\'{}\' is not a integer. Check that the value {} doesn’t contain decimal points or any characters other than 0-9. Integer values should not be in quotes. ', # noqa
-                           'number': '\'{}\' is not a number. Check that the value {} doesn’t contain any characters other than 0-9 and dot (\'.\'). Number values should not be in quotes. ', # noqa
-                           'object': '\'{}\' is not a JSON object',
-                           'array': '\'{}\' is not a JSON array'}
+validation_error_template_lookup = {
+    'date': 'Date is not in the correct format. The correct format is YYYY-MM-DD.',
+    'date-time': 'Date is not in the correct format. The correct format is YYYY-MM-DDThh:mm:ssZ.',
+    'uri': 'Invalid uri found',
+    'string': '\'{}\' should be a string. Check that the value {} has quotes at the start and end. Escape any quotes in the value with \'\\\'', # noqa
```

Comment:
Quoting a non-string makes it look like a string, thus making it harder to see what's wrong… The quote characters should be removed.

One alternative: If the value is a string, first wrap it in double quotes. Then, for the web UI, wrap that value in `<code>` tags before performing the string substitution. For other UIs, perhaps wrap that value in backticks instead of single quotes.

This same comment applies to all other similar messages. Quoting a non-integer makes it look like a string, so a user might think they have a string where an integer should be.

Reply:
Hmm, looking at the web UI I see fewer single-quote characters than in these messages.

That said, I notice that string values are not wrapped in double-quotes, which makes them harder to identify as strings – so at least that part of my comment is relevant.